### PR TITLE
Add support for vcpu ms

### PR DESCRIPTION
**Pull request by guybedford**
Resolves https://github.com/fastly/js-compute-runtime/issues/925.

This is implemented as a new `fastly:runtime` interface, exporting a `vCpuTime()` function.

### CODE REVIEW

File: .vscode/c_cpp_properties.json [hunk 1]
```diff
@@ -3,12 +3,11 @@
...
-                "${workspaceFolder}/runtime/spidermonkey/debug/include",
-                "${workspaceFolder}/runtime/spidermonkey/release/include",
+                "${workspaceFolder}/runtime/StarlingMonkey/include",
+                "${workspaceFolder}/runtime/StarlingMonkey/builtins/web",
+                "${workspaceFolder}/runtime/fastly/build-debug/",
                 "/opt/wasi-sdk/share/wasi-sysroot/include/",
-                "${workspaceFolder}/runtime/js-compute-runtime",
-                "${workspaceFolder}/runtime/js-compute-runtime/build/openssl-3.0.7/include",
-                "${workspaceFolder}/runtime/js-compute-runtime/third_party/fmt/include"
+                "${workspaceFolder}/runtime/fastly/host-api"
```
Comment (guybedford) [inline .vscode/c_cpp_properties.json:10]: I'm trying to work out how to get intellisense to configure properly, but am still not finding this working. Any suggestions very much welcome, but at least this is a step in the right direction.

File: integration-tests/js-compute/fixtures/app/src/runtime.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  ok(cpuTime > 0);
+  ok(cpuTime < 1000);
+  const arr = [];
+  for (let i = 0; i < 1000; i++) {
+    arr.push(i);
+  }
+  const cpuTime2 = vCpuTime();
+  ok(cpuTime2 > cpuTime);
+  ok(cpuTime2 - cpuTime > 1);
+  ok(cpuTime2 - cpuTime < 1000);
```
Comment (elliottt) [inline integration-tests/js-compute/fixtures/app/src/runtime.js:21]: The two comparisons around the upper bound of `1000` might end up being flaky tests. Does it make sense to bump it up a bit more?

**Review by elliottt — APPROVED** [review]
Looks good! Just one comment about the test possibly being flaky.

File: .vscode/c_cpp_properties.json (hunk 1 above)
Comment (elliottt) [reply .vscode/c_cpp_properties.json:10]: `cmake` can generate a `compile_commands.json`, which might be more reliable. Then you can configure `clangd` to read that instead.

File: integration-tests/js-compute/fixtures/app/src/runtime.js (hunk 2 above)
Comment (guybedford) [reply integration-tests/js-compute/fixtures/app/src/runtime.js:21]: It's denoted in milliseconds if that helps to clarify?

File: .vscode/c_cpp_properties.json (hunk 1 above)
Comment (guybedford) [reply .vscode/c_cpp_properties.json:10]: Amazing thank you, this worked.

File: integration-tests/js-compute/fixtures/app/src/runtime.js (hunk 2 above)
Comment (elliottt) [reply integration-tests/js-compute/fixtures/app/src/runtime.js:21]: Yep, I'm just worried that the test will drift in execution time, and cause failures if somehow it goes beyond one second. If you're confident that we won't see that, then we can go ahead and merge.

Comment (guybedford) [reply integration-tests/js-compute/fixtures/app/src/runtime.js]: I've bumped it to 3000.

**Pull request merged**